### PR TITLE
Add Support for Sensirion SFA30 sensor

### LINKED ISSUE #1926: SFA30 Support
**Add Support for Sensirion SFA30 sensor**

https://github.com/Sensirion/arduino-i2c-sfa3x
https://sensirion.com/products/catalog/SFA30/

### PR DESCRIPTION
**Pull request by ghsensdev**
# What does this implement/fix?

This implements the SFA30 Formaldehyde Sensor.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1926

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#3260

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```
sensor:
  - platform: sfa30
    formaldehyde:
      name: "Formaldehyde"
    temperature:
      name: "Temperature"
    humidity:
      name: "Humidity"
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/sfa30/sensor.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+async def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(var, config)
+    await i2c.register_i2c_device(var, config)
+
+    for key, funcName in SENSOR_MAP.items():
+        if key in config:
+            sens = await sensor.new_sensor(config[key])
```
Comment (jesserockz) [inline esphome/components/sfa30/sensor.py]:
```suggestion
        if sensor_config := config.get(key):
            sens = await sensor.new_sensor(sensor_config)
```

**Review by jesserockz — DISMISSED** [review]
Once minor change below, otherwise looks good. THanks

**Comment by ghsensdev** [comment]
Do I need to update the branch because out of date with the base branch?
With merge commit or with rebase?

**Review by jesserockz — APPROVED** [review]

**Comment by jesserockz** [comment]
@ghsensdev no its fine, as long as there are no conflicts we merge with squash

**Pull request merged**